src/box/iproto.cc

```diff
@@ -3619,6 +3637,13 @@ iproto_send_restart_msg(void)
 int
 iproto_listen(const struct uri_set *uri_set)
 {
+	int ret = -1;
+	struct errinj *inj = NULL;
+	latch_lock(&shutdown_latch);
```

Comment:
It's unclear why you take a lock here and hold during the whole function execution. It's unclear why you do *not* take the lock in `iproto_session_new`. This requires comments.

Reply:
In new version latch is dropped and instead we handle attempt to listen again in iproto thread as discussed f2f.